modules/internal/ChapelBase.chpl

```diff
@@ -1237,6 +1240,9 @@ module ChapelBase {
     }
     if countRunningTasks {
       here.runningTaskCntAdd(1);  // decrement is in _waitEndCount()
+      chpl_comm_task_create();    // countRunningTasks is a proxy for "is local"
+                                  // here.  Comm layers are responsible for the
+                                  // remote case themselves.
```

Comment:
Just confirming -- this means that the comm layer will fence before doing an on-stmt, right?

Reply:
Yes.